fix protocol documentation (#319)
rsocket/rsocket: Protocol.md

## Status

This protocol is currently a __draft__ for the final specifications. 
Current version of the protocol is __0.2__ (Major Version: __0__, Minor Version: __2__). This is currently considered a 1.0 Release Candidate. 

> Final testing is being done in Java and C++ implementations with goal to release 1.0 in the near future. 

Reviewer: Let's remove this dated line altogether.

Author: Should I also remove the `Status` Heading ?

## Introduction

> RSocket, as the name implies, is a socket with the [Reactive Streams](http://www.reactive-streams.org/) semantics, that is where the 'R' comes from, across an asynchronous, binary boundary. 

Reviewer: ```suggestion
RSocket is an application protocol that provides [Reactive Streams](http://www.reactive-streams.org/) semantics across a network, asynchronous, binary boundary. 
```
I don't think we should lose the term "application protocol".

## Data And Metadata

> RSocket provides mechanisms for applications to distinguish a payload into two types. Data and Metadata, this distinction is left to the application.

Reviewer: ```suggestion
RSocket provides mechanisms for applications to differentiate between two types of content within a payload, Data and Metadata, with the distinction left to applications to make use of according to their needs.
```

The RSocket protocol uses a lower level transport protocol to carry RSocket frames. A transport protocol MUST provide the following:

> 1. Unicast [Reliable Delivery](https://en.wikipedia.org/wiki/Reliability_(computer_networking)) Delivery. Meaning that it's a one-to-one transmission from one point in the network to another point; that is, one sender and one receiver. In addition to being "at least once", i.e. at least one copy of the message is guaranteed to be delivered to the recipient.

Reviewer: ```suggestion
1. Unicast [Reliable Delivery](https://en.wikipedia.org/wiki/Reliability_(computer_networking)), i.e. between a single sender and a single receiver with "at least once" message delivery guarantees.
```
"Delivery" is duplicated (once i the link and once after the link).

> 1. [Connection-Oriented](https://en.wikipedia.org/wiki/Connection-oriented_communication) along with preservation of frame ordering. if Frame A is sent by the same source as Frame B, then Frame A will always arrive before Frame B. No assumptions about ordering across sources is assumed.

Reviewer: "if" should be capitalized (start of sentence).

1) If Stream ID re-use is not employed:

    - No new streams can be created, thus a new connection MUST be established to create new streams once the max has been met. 

2) If Stream IDs are re-used:

    - The Requester MUST re-use IDs by wrapping and restarting at ID 1 for client, and ID 2 for server, and incrementing sequentially by 2s as stated above.
    - The Requester MUST skip IDs still in use. 
    - The Responder MAY choose to ERROR[REJECT] any Stream ID if it considers the ID still in use. The Requester MAY retry on the next sequential Stream ID considered unused.
    - If all Stream IDs are concurrently in use, no new streams can be created, thus a new connection MUST be established to create new streams.

It is RECOMMENDED that Stream ID re-use only be used in combination with resumability. 

### Frame Types

|  Type                          | Value  | Description |
|:-------------------------------|:-------|:------------|

> | __RESERVED__                                     | 0x00 | __Reserved__ for any operation involving the connection. |

Reviewer 2: This is not correct.  `Stream ID of value __0__` but this section is discussing the frame types, not stream ids.

Author: oh !!!! got it
```suggestion
| __RESERVED__                                     | 0x00 | __Reserved__ |
```

Author: oh !!! got it 👍🏽
